tests/codeception/acceptance/SubThemeCest.php

```diff
@@ -0,0 +1,133 @@
+   */
+  public function __construct() {
+    $this->themeName = Factory::create()->firstName;
+    $this->themePath = realpath(dirname(drupal_get_path('theme', 'stanford_basic'))) . '/' . strtolower($this->themeName);
+  }
+
+  /**
+   * Before the tests, create a stubbed out subtheme.
+   *
+   * @param \AcceptanceTester $I
+   *   Tester.
+   */
+  public function _before(AcceptanceTester $I) {
+    $this->createTheme();
+  }
+
+  /**
+   * Always cleanup the config after testing.
+   *
+   * @param \AcceptanceTester $I
+   *   Tester.
+   */
+  public function _after(AcceptanceTester $I) {
+    $this->_failed($I);
+  }
+
+  /**
+   * Always cleanup the config after testing.
+   *
+   * @param \AcceptanceTester $I
+   *   Tester.
+   */
+  public function _failed(AcceptanceTester $I) {
+    $this->runConfigImport($I, TRUE);
+    unlink($this->themePath . '/' . strtolower($this->themeName) . '.info.yml');
+    rmdir($this->themePath);
+  }
+
+  /**
+   * Enable the subtheme and the config should reflect the changes done.
+   */
+  public function testSubTheme(AcceptanceTester $I) {
+    $I->runDrush('theme:enable -y ' . strtolower($this->themeName));
+    $I->logInWithRole('administrator');
+    $I->amOnPage('/admin/appearance');
+    $I->click('Set as default', sprintf('a[title="Set %s as default theme"]', $this->themeName));
+    $I->amOnPage('/');
+    $I->canSeeResponseCodeIs(200);
+
+    $this->runConfigImport($I);
+    $result = $I->runDrush('config-get system.theme default --format=json');
+    $result = json_decode($result, TRUE);
+    $I->assertEquals(strtolower($this->themeName), $result['system.theme:default']);
+
+    $I->amOnPage('/admin/appearance');
+    $I->click('Set as default', sprintf('a[title="Set %s as default theme"]', 'Stanford Basic'));
+    $this->runConfigImport($I);
+
+    $result = $I->runDrush('config-get system.theme default --format=json');
+    $result = json_decode($result, TRUE);
+    $I->assertEquals('stanford_basic', $result['system.theme:default']);
+
+    $I->amOnPage('/');
+    $I->canSeeResponseCodeIs(200);
+  }
+
+  /**
+   * Run config import and adjust saml module if necessary.
+   *
+   * @param \AcceptanceTester $I
+   *   Tester.
+   * @param bool $disable_config_ignore
+   *   If config ignore module should be disabled first.
+   */
+  protected function runConfigImport(AcceptanceTester $I, $disable_config_ignore = FALSE) {
+    $drush_response = $I->runDrush('pm-list --filter=name=stanford_ssp --format=json');
+    $drush_response = json_decode($drush_response, TRUE);
+    $saml_enabled = $drush_response['stanford_ssp']['status'] == 'Enabled';
+
+    if ($disable_config_ignore) {
+      $I->runDrush('pmu config_ignore');
+    }
+
+    $I->runDrush('config-import -y');
+    if (!$saml_enabled) {
+      $I->runDrush('pm-uninstall simplesamlphp_auth -y');
+    }
+  }
+
+  /**
+   * Create a stub of a subtheme based on stanford_basic.
+   */
+  protected function createTheme() {
```

Comment:
Why not just package a test theme with this profile like the unit tests do? Could save issues with systems with file permission or write issues.

Reply:
This way it creates a new theme name very test. So instead of verifying only 1 scenario, it tests a different scenario every time.

Reply:
![https://media3.giphy.com/media/xUA7aUNw61j9Vdzs0U/giphy.gif?cid=ecf05e47a95ba32a1c98890a4251dcb9cda7a3f9d335c695&rid=giphy.gif](https://media3.giphy.com/media/xUA7aUNw61j9Vdzs0U/giphy.gif?cid=ecf05e47a95ba32a1c98890a4251dcb9cda7a3f9d335c695&rid=giphy.gif)